Arithmetical puzzles

lmlearning opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

## Eval details 📑
### Eval name
Arithmetical Puzzles

### Eval description

This Eval evaluates performance on arithmetical puzzles expressed in natural language. The examples are taken from historical puzzle books that are out of copyright and the question and answers have been adapted to allow clear prompting and answers for evaluation. 

### What makes this a useful eval?

Performance on math problems can be a particular pain point for LLMs and this eval is no exception. In my initial testing, GPT-3.5 gets ~6% accuracy and GPT-4 (evaluated via the web interface) < 20%. Therefore, it is useful as a check for improving mathematical reasoning abilities.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [ x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [x ] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [ x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [ x] **Include at least 15 high quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

This is a challenging test, also for humans. Some of the puzzles require careful thought and others effectively a brute force search to come up with a solution. Therefore it tests arithmetical puzzle solving better than many benchmarks.

## Eval structure 🏗️

Your eval should
- [ x] Check that your data is in `evals/registry/data/{name}`
- [ x] Check that your yaml is registered at `evals/registry/evals/{name}.yaml`
- [x ] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [ x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [x ] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x ] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [x ] I have filled out all required fields in the evals PR form
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "Tommy: \"How old are you, mamma?\" Mamma: \"Let me think, Tommy. Well, our three ages add up to exactly seventy years.\" Tommy: \"That's a lot, isn't it? And how old are you, papa?\" Papa: \"Just six times as old as you, my son.\" Tommy: \"Shall I ever be half as old as you, papa?\" Papa: \"Yes, Tommy; and when that happens our three ages will add up to exactly twice as much as to-day.\" Tommy: \"And supposing I was born before you, papa; and supposing mamma had forgot all about it, and hadn't been at home when I came; and supposing\" Mamma: \"Supposing, Tommy, we talk about bed. Come along, darling. You'll have a headache.\" Now, if Tommy had been some years older he might have calculated the exact ages of his parents from the information they had given him. Can you find out the exact age of mamma in months? Please give only the numerical answer as a single integer indicating the number of months. Do not give any reasoning or logic for your answer."}], "ideal": "350"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "\"My husband's age,\" remarked a lady the other day, \"is represented by the figures of my own age reversed. He is my senior, and the difference between our ages is one-eleventh of their sum.\" What is the husband's age? Please give only the numerical answer as a single integer indicating the number of years. Do not give any reasoning or logic for your answer."}], "ideal": "54"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "When the Smileys recently received a visit from the favourite uncle, the fond parents had all the five children brought into his presence. First came Billie and little Gertrude, and the uncle was informed that the boy was exactly twice as old as the girl. Then Henrietta arrived, and it was pointed out that the combined ages of herself and Gertrude equalled twice the age of Billie. Then Charlie came running in, and somebody remarked that now the combined ages of the two boys were exactly twice the combined ages of the two girls. The uncle was expressing his astonishment at these coincidences when Janet came in. \"Ah! uncle,\" she exclaimed, \"you have actually arrived on my twenty-first birthday!\" To this Mr. Smiley added the final staggerer: \"Yes, and now the combined ages of the three girls are exactly equal to twice the combined ages of the two boys.\" Please give the age of Billie. Please give only the numerical answer as a single integer indicating the number of months. Do not give any reasoning or logic for your answer."}], "ideal": "42"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "Edwin: \"Do you know, when the Timpkinses married eighteen years ago Timpkins was three times as old as his wife, and to-day he is just twice as old as she?\" Angelina: \"Then how old was Mrs. Timpkins on the wedding day?\" Can you answer Angelina's question? Please give the age of Mrs. Timpkins. Please give only the numerical answer as a single integer indicating the number of years. Do not give any reasoning or logic for your answer."}], "ideal": "18"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "\"Mother, I wish you would give me a bicycle,\" said a girl of twelve the other day. \"I do not think you are old enough yet, my dear,\" was the reply. \"When I am only three times as old as you are you shall have one.\" Now, the mother's age is forty-five years. When may the young lady expect to receive her present? Please give the length of time as a single integer indicating the number of months. Do not give any reasoning or logic for your answer."}], "ideal": "54"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "There were two families living next door to one another at Tooting Becthe Jupps and the Simkins. The united ages of the four Jupps amounted to one hundred years, and the united ages of the Simkins also amounted to the same. It was found in the case of each family that the sum obtained by adding the squares of each of the children's ages to the square of the mother's age equalled the square of the father's age. In the case of the Jupps, however, Julia was one year older than her brother Joe, whereas Sophy Simkin was two years older than her brother Sammy. What was the age of Mr. Jupp? Please give the age as a single integer indicating the number of years. Do not give any reasoning or logic for your answer."}], "ideal": "39"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "Write eleven thousand eleven hundred and eleven. Please give the answer as a single integer. Do not give any reasoning or logic for your answer."}], "ideal": "12111"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "In a recent motor ride it was found that we had gone at the rate of ten miles an hour, but we did the return journey over the same route, owing to the roads being more clear of traffic, at fifteen miles an hour. What was our average speed? Please give the answer as an integer. Do not give any reasoning or logic for your answer. \n"}], "ideal": "12"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "Two trains start at the same time, one from London to Liverpool, the other from Liverpool to London. If they arrive at their destinations one hour and four hours respectively after passing one another, how much faster is one train running than the other? Please give the answer as an integer, indicating the ratio. Do not give any reasoning or logic for your answer."}], "ideal": "2"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "I set out the other day to ride in a motor-car from Acrefield to Butterford, but by mistake I took the road going via Cheesebury, which is nearer Acrefield than Butterford, and is twelve miles to the left of the direct road I should have travelled. After arriving at Butterford I found that I had gone thirty-five miles. What is the distance from Acrefield to Butterford? Please give the answer as an integer. Do not give any reasoning or logic for your answer. \n"}], "ideal": "25"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "Sir Edwyn de Tudor was going to rescue his lady-love, the fair Isabella, who was held a captive by a neighbouring wicked baron. Sir Edwyn calculated that if he rode fifteen miles an hour he would arrive at the castle an hour too soon, while if he rode ten miles an hour he would get there just an hour too late. Now, it was of the first importance that he should arrive at the exact time appointed, in order that the rescue that he had planned should be a success, and the time of the tryst was five o'clock, when the captive lady would be taking her afternoon tea. How many miles did Sir Edwyn de Tudor have to ride. Please give the answer as an integer. Do not give any reasoning or logic for your answer. \n"}], "ideal": "60"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "A man had a basket containing fifty potatoes. He proposed to his son, as a little recreation, that he should place these potatoes on the ground in a straight line. The distance between the first and second potatoes was to be one yard, between the second and third three yards, between the third and fourth five yards, between the fourth and fifth seven yards, and so onan increase of two yards for every successive potato laid down. Then the boy was to pick them up and put them in the basket one at a time, the basket being placed beside the first potato. How far would the boy have to travel to accomplish the feat of picking them all up? We will not consider the journey involved in placing the potatoes, so that he starts from the basket with them all laid out. Please give the answer as an integer indicating the number of yards. Do not give any reasoning or logic for your answer."}], "ideal": "80850"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "Using each figure once, and only once, we can form two multiplication sums that have the same product, and this may be done in many ways. For example, 7x658 and 14x329 contain all the digits once, and the product in each case is the same4,606. Now, it will be seen that the sum of the digits in the product is 16, which is neither the highest nor the lowest sum so obtainable. Can you find the solution of the problem that gives the lowest possible sum of digits in the common product? Please give the answer as an integer. Do not give any reasoning or logic for your answer. "}], "ideal": "4002"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "If I multiply 51,249,876 by 3 (thus using all the nine digits once, and once only), I get 153,749,628 (which again contains all the nine digits once). Similarly, if I multiply 16,583,742 by 9 the result is 149,253,678, where in each case all the nine digits are used. Now, take 6 as your multiplier and try to arrange the remaining eight digits so as to produce by multiplication a number containing all nine once, and once only. What is the result? Please give the answer as an integer. Do not give any reasoning or logic for your answer."}], "ideal": ["195287346","195,287,346"]}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "In a certain mixed school, where a special feature was made of the inculcation of good manners, they had a curious rule on assembling every morning. There were twice as many girls as boys. Every girl made a bow to every other girl, to every boy, and to the teacher. Every boy made a bow to every other boy, to every girl, and to the teacher. In all there were nine hundred bows made in that model academy every morning. Now, can you say exactly how many boys there were in the school? Please give the answer as an integer. Do not give any reasoning or logic for your answer."}], "ideal": "10"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "\"A man I know,\" said Teddy Nicholson at a certain family party, \"possesses a string of thirty-three pearls. The middle pearl is the largest and best of all, and the others are so selected and arranged that, starting from one end, each successive pearl is worth 100 more than the preceding one, right up to the big pearl. From the other end the pearls increase in value by 150 up to the large pearl. The whole string is worth 65,000. What is the value of that large pearl?\" Please give the answer as an integer. Do not give any reasoning or logic for your answer."}], "ideal": "3000"}
{"input": [{"role": "system", "content": "Complete the phrase as concisely as possible."}, {"role": "user", "content": "Tim Murphy and Pat Donovan were engaged by the local authorities to paint the lamp-posts in a certain street. Tim, who was an early riser, arrived first on the job, and had painted three on the south side when Pat turned up and pointed out that Tim's contract was for the north side. So Tim started afresh on the north side and Pat continued on the south. When Pat had finished his side he went across the street and painted six posts for Tim, and then the job was finished. As there was an equal number of lamp-posts on each side of the street, the simple question is: How many more did the man who painted the most lamp-posts paint than the other?  Please give the answer as an integer. Do not give any reasoning or logic for your answer. \n"}], "ideal": "6"}

 
  ```
</details>

Ein-Tim commented on .gitattributes:
Please don't change this file.

Ein-Tim commented on .gitattributes:
```diff
@@ -1 +0,0 @@
-evals/registry/data/**/*.jsonl filter=lfs diff=lfs merge=lfs -text
```
Please simply revert the changes to this file. Do not change it and do not delete it. Thanks.

lmlearning replied:
Done now, I think

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request some changes.

1. You need to only push the `jsonl` file as a git lfs file. You've also pushed the `.yaml` file as lfs. Kindly push the `.yaml` file as a normal file.

2. In some samples, there are unnecessary unicode characters. Kindly remove those characters. Here is a part of a sample for reference:

    ```
    Tommy: \"And supposing I was born before you, papa; and supposing mamma had forgot all about it, and hadn't been at home when I came; and supposing\u00e2\u20ac\u201d\u00e2\u20ac\u201d\" Mamma: \"Supposing, Tommy, we talk about bed. Come along, darling. You'll have a headache.\"
    ```

3. Merge the `main` branch into your branch to resolve workflow-related issues.

We would love to review the PR again after the suggested changes.

lmlearning commented:
Thank you for the guidance. It is much appreciated. I've removed the yaml file from LFS, removed the unicode characters and merged the main branch into my forked branch as requested. Hope that resolves the errors.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.
